tests/Tests.AzureAppConfiguration/Integration/IntegrationTests.cs

```diff
@@ -0,0 +1,1923 @@
+        private readonly HashSet<string> _createdConfigKeys = new HashSet<string>();
+        private readonly HashSet<string> _createdSecretNames = new HashSet<string>();
+        private readonly HashSet<string> _createdSnapshotNames = new HashSet<string>();
+
+        /// <summary>
+        /// Gets a DefaultAzureCredential for authentication.
+        /// </summary>
+        private DefaultAzureCredential GetCredential()
+        {
+            return new DefaultAzureCredential(new DefaultAzureCredentialOptions
+            {
+                ExcludeSharedTokenCacheCredential = true
+            });
+        }
+
+        /// <summary>
+        /// Gets the current subscription ID by reading from the JSON file created by the PowerShell script.
+        /// NOTE: The PowerShell script must be run manually before running the tests.
+        /// </summary>
+        private string GetCurrentSubscriptionId()
+        {
+            string subscriptionIdFromEnv = Environment.GetEnvironmentVariable("AZURE_SUBSCRIPTION_ID");
+
+            if (!string.IsNullOrEmpty(subscriptionIdFromEnv))
+            {
+                return subscriptionIdFromEnv;
+            }
+
+            // Read the JSON file created by the script
+            string jsonPath = Path.Combine(AppDomain.CurrentDomain.BaseDirectory, "Integration", SubscriptionJsonPath);
+
+            if (!File.Exists(jsonPath))
+            {
+                throw new InvalidOperationException($"Subscription JSON file not found at {jsonPath}. Run the GetAzureSubscription.ps1 script first.");
+            }
+
+            string jsonContent = File.ReadAllText(jsonPath);
+
+            using JsonDocument doc = JsonDocument.Parse(jsonContent);
+            JsonElement root = doc.RootElement;
+
+            bool success = root.GetProperty("Success").GetBoolean();
+
+            if (!success)
+            {
+                throw new InvalidOperationException(root.GetProperty("ErrorMessage").GetString());
+            }
+
+            return root.GetProperty("SubscriptionId").GetString();
+        }
+
+        /// <summary>
+        /// Returns the connection string for connecting to the app configuration store.
+        /// </summary>
+        private string GetConnectionString()
+        {
+            return _connectionString;
+        }
+
+        /// <summary>
+        /// Creates a unique prefix for test keys to ensure test isolation
+        /// </summary>
+        private string GetUniqueKeyPrefix(string testName)
+        {
+            // Use a combination of the test prefix and test method name to ensure uniqueness
+            return $"{TestKeyPrefix}-{testName}-{Guid.NewGuid().ToString("N").Substring(0, 8)}";
+        }
+
+        /// <summary>
+        /// Creates a snapshot with the given name containing the test context's settings
+        /// </summary>
+        private async Task<string> CreateSnapshot(string snapshotName, TestContext testContext)
```

Comment:
Most of `TestContext` isn't used here. I'd like to see the caller pass what's needed only. In this case I'd suggest passing `IEnumerable<ConfigurationSettingsFilter>`